lib/usd/ui/views/ItemDelegate.cpp

```diff
@@ -0,0 +1,260 @@
+//
+// Copyright 2019 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "ItemDelegate.h"
+#include "TreeModel.h"
+#include "TreeItem.h"
+
+#include <QtWidgets/QComboBox>
+#include <QtWidgets/QLabel>
+#include <QtGui/QPainter>
+#include <QtWidgets/QVBoxLayout>
+#include <QtWidgets/QHBoxLayout>
+#include <QtCore/QSortFilterProxyModel>
+
+#include <pxr/usd/usd/prim.h>
+#include <pxr/usd/usd/variantSets.h>
+
+#include <cassert>
+
+MAYAUSD_NS_DEF {
+
+QWidget* ItemDelegate::createEditor(QWidget* parent, const QStyleOptionViewItem& option, const QModelIndex& index) const
+{
+	// Check out special type role (instead of column) since it is only set then this
+	// prim has a variant set and variant selection.
+	int type = index.data(kTypeRole).toInt();
+	if (type == kVariants)
+	{
+		TreeItem* treeItem = getTreeItemForIndex(index);
+		assert(treeItem != nullptr);
+		if (nullptr != treeItem)
```

Comment:
assert is disabled when NDEBUG is defined (which it is in the RelWithDebInfo build). So in a RelWithDebInfo build the assert will do nothing. In a debug build (which we use internally), if the assert fires it will print a message and then call abort. That is what we want so we will be forced to investigate why it was null.